osgp/protocol-adapter-oslp/oslp/src/main/java/org/opensmartgridplatform/oslp/OslpDecoder.java

```diff
@@ -46,80 +45,99 @@ public OslpDecoder(final String signature, final String provider) {
 
     @Override
     protected void decode(final ChannelHandlerContext ctx, final ByteBuf in, final List<Object> out) throws Exception {
-        LOGGER.debug("Decoding state: {}", this.state());
+        final DecodingState decodingState = this.state();
+        LOGGER.debug("Decoding state: {}", decodingState);
 
-        if (this.state().compareTo(DecodingState.SECURITY_KEY) == 0) {
+        switch (decodingState) {
+        case SECURITY_KEY:
             LOGGER.debug("Decoding security key.");
             this.decodeSecurityKey(in);
             this.checkpoint(DecodingState.SEQUENCE_NUMBER);
-        }
-
-        if (this.state().compareTo(DecodingState.SEQUENCE_NUMBER) <= 0) {
+            // fall-through
+        case SEQUENCE_NUMBER:
             LOGGER.debug("Decoding sequence number.");
             this.decodeSequenceNumber(in);
             this.checkpoint(DecodingState.DEVICE_ID);
-        }
-
-        if (this.state().compareTo(DecodingState.DEVICE_ID) <= 0) {
+            // fall-through
+        case DEVICE_ID:
             LOGGER.debug("Decoding device id.");
             this.decodeDeviceId(in);
             this.checkpoint(DecodingState.LENGTH_INDICATOR);
-        }
-
-        if (this.state().compareTo(DecodingState.LENGTH_INDICATOR) <= 0) {
+            // fall-through
+        case LENGTH_INDICATOR:
             LOGGER.debug("Decoding length indicator.");
             this.decodeLengthIndicator(in);
             this.checkpoint(DecodingState.PAYLOAD_MESSAGE);
-        }
-
-        if (this.state().compareTo(DecodingState.PAYLOAD_MESSAGE) <= 0) {
+            // fall-through
+        case PAYLOAD_MESSAGE:
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce this switch case number of lines from 6 to at most 5, for example by extracting code into methods. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://ci.opensmartgridplatform.org/sonarqube/coding_rules#rule_key=squid%3AS1151)